debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlChangeEventSourceFactory.java

```diff
@@ -85,8 +89,35 @@ private void preSnapshot() {
     }
 
     private void modifyAndFlushLastRecord(Function<SourceRecord, SourceRecord> modify) throws InterruptedException {
```

Comment:
This likely applies to MariaDB. I might suggest we create a `BinlogChangeEventSourceFactory` abstract base class that both MySQL and MariaDB extend from, and put this method in the abstract base class.